### PR TITLE
8255378: [Vector API] Remove redundant vector length check after JDK-8254814 and JDK-8255210

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

As @iwanowww pointed out [1] that there are redundant vector length checks for reductionI [2] and reductionS [3].
It would be better to remove them.

Testing:
 - jdk/incubator/vector on both AVX512 and AVX256 machines

Thanks.
Best regards,
Jie

[1] https://github.com/openjdk/jdk/pull/791#discussion_r510687005
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86.ad#L4429
[3] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86.ad#L4625

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x32 | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |  ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8255378](https://bugs.openjdk.java.net/browse/JDK-8255378): [Vector API] Remove redundant vector length check after JDK-8254814 and JDK-8255210

### Reviewers
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**)
 * [Andrew Dinn](https://openjdk.java.net/census#adinn) (@adinn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/854/head:pull/854`
`$ git checkout pull/854`

### CODE REVIEW

**Review by iwanowww — APPROVED** [review]
Looks good.

**Comment by DamonFool** [comment]
> Looks good.

Thanks @iwanowww for your review.
May I get a second review for this change?
Thanks.

**Review by adinn — APPROVED** [review]
This is arguably trivial and needs no review but here is one anyway.

**Comment by DamonFool** [comment]
> This is arguably trivial and needs no review but here is one anyway.

Thanks @adinn for your review.
Pushed.

/integrate

**Pull request closed without merging**
